### ISSUE TITLE
S3 output plugin not supporting "format" option

### ISSUE THREAD

**Issue by clebio** [elastic/logstash#4203]
I'm having troubling getting Logstash to write to S3, and get the following error while grasping for straws:

`{:timestamp=>"2015-11-16T23:34:17.355000+0000", :message=>"Unknown setting 'format' for s3", :level=>:error}`

I'm using the Debian Logstash package:

```
# /opt/logstash/bin/logstash --version
logstash 2.0.0
```

Using an IAM role (where `aws s3 ls s3://...` works fine on the host server, given I install the awscli package from apt).

My `/etc/logstash/logstash.conf` is dead-simple:

```
input {
  beats {
    port => 5044
  }
}

output {
    file { path => "/var/log/logstash/messages.log" }
    s3 {
       bucket => "redacted"
       prefix => "logstash/"
       format => "plain"
    }
}
```

Logstash is receiving JSON messages from the filebeat input just file. And the "file" output is working great as well. I'm at a loss as to how I can debug this further.

**Comment by clebio** [comment]
I should add that the `format` option is provided as a valid S3 output configuration on the Elastic docs: https://www.elastic.co/guide/en/logstash/current/plugins-outputs-s3.html

I've been finding various discrepancies between resolved issues here, the official docs, and the actual behavior of my 2.0 Logstash installation. I'll try to share the other issues that have surfaced as I remember them.

**Comment by jordansissel** [comment]
Looks like the docs are wrong.

**Comment by clebio** [comment]
Indeed, it seems the docs have remained, yet the config options doesn't exist under the 2.0.2 branch:
https://github.com/ph/logstash-output-s3/blob/master/lib/logstash/outputs/s3.rb#L140
versus
https://github.com/ph/logstash-output-s3/blob/v2.0.2/lib/logstash/outputs/s3.rb#L70-L97

More generally, my Logstash server is receiving logs from a filebeat input, and writing to local file, but not to S3. When I start Logstash manually, with verbose logging, it appears to be setting up the S3 connection fine. Is there some way to get yet-further detailed logging?

```
# /opt/logstash/bin/logstash -f /etc/logstash.conf  --verbose
Using version 0.9.x input plugin 'beats'. This plugin should work but would benefit from use by folks like you. Please let us know if you find bugs or have suggestions on how to improve this plugin. {:level=>:info}
Default settings used: Filter workers: 1
Starting Beats input listener {:address=>"0.0.0.0:5044", :level=>:info}
Registering s3 output {:bucket=>"redacted", :endpoint_region=>"us-east-1", :level=>:info}
Pipeline started {:level=>:info}
Logstash startup completed
```

**Comment by clebio** [comment]
Well, a little progress. If I set `time_file` to something (say, 1), it starts writing to S3. [The docs](https://www.elastic.co/guide/en/logstash/current/plugins-outputs-s3.html#plugins-outputs-s3-time_file) are pretty cryptic on this (and not grammatical):

> Set the time, in minutes, to close the current sub_time_section of bucket. If you define file_size you have a number of files in consideration of the section and the current tag. 0 stay all time on listerner, beware if you specific 0 and size_file 0, because you will not put the file on bucket, for now the only thing this plugin can do is to put the file when logstash restart.

Which to me would indicate that I need to specify *either of* `size_file` or `time_file` -- and yet, if I set `size_file` (e.g. 1024) but *not* `time_file`, nothing is written to S3. And the same docs don't say any such thing under [size_file](https://www.elastic.co/guide/en/logstash/current/plugins-outputs-s3.html#plugins-outputs-s3-size_file).

**Comment by edifecs-pankaj** [comment]
So what shall be the correct/ideal values for size_file or time_file as in my case, i see log of files are created in temp folder and they are mostly pushed to s3 bucket only when i restart logstash.
